language/Type-Classes.md

```diff
@@ -137,36 +137,128 @@ See also the section in [PureScript by Example](https://book.purescript.org/chap
 
 ## Type Class Deriving
 
-Some type class instances can be derived automatically by the PureScript compiler. To derive a type class instance, use the `derive instance` keywords:
+The compiler can derive type class instances to spare you the tedium of writing boilerplate. There are a few ways to do this depending on the specific Type and Class being derived.
 
-```purescript
-newtype Person = Person { name :: String, age :: Int }
+### Derive from `newtype`
+
+If you'd like to use a wrapped type's instance for your newtype, use the `derive newtype` keywords.
+
+For example, let's say you want to add two `Points` values using the `Semiring` instance of the wrapped `Int`.
```

Comment:
I think the `semiring` example is a great example of something useful that (iirc) can't be derived via the usual `derive` keyword and has to use newtype deriving. However, the example that I think best illustrates what exactly newtype deriving is doing is `Show`.

Specifically:

```purs
newtype Point = Point Int

derive newtype instance showPoint :: Show Point
```

and then in the repl:

```
> show (Point 1)
"1"
```

This illustrates that you're really using the show instance for `Int` -- you don't see the "Point" constructor at all. Same idea applies to things like deriving a `DecodeJson` instance (which I don't think we should cover here).

I don't think that we need to delete the semiring example, but maybe we could also touch on the show example to demonstrate this further.

Reply:
> you don't see the "Point" constructor at all.

This is clearer to contrast if the reader is already familiar with `generic`, which I believe is the only `derive` option that prints the constructor. I can still include a lookahead note. Wondering if there's another example that doesn't depend on `generic`.